plonk-core/src/transcript.rs

```diff
@@ -7,91 +7,76 @@
 //! This is an extension over the [Merlin Transcript](Transcript) which adds a
 //! few extra functionalities.
 
-use ark_ec::PairingEngine;
 use ark_ff::{Field, PrimeField};
-use ark_poly_commit::kzg10::Commitment;
+use ark_poly::univariate::DensePolynomial;
+use ark_poly_commit::{LabeledCommitment, PolynomialCommitment};
 use ark_serialize::CanonicalSerialize;
 use core::marker::PhantomData;
 use merlin::Transcript;
 
-/// Wrapper around [`Transcript`]
-#[derive(derivative::Derivative)]
-#[derivative(Clone)]
-pub struct TranscriptWrapper<E>
-where
-    E: PairingEngine,
-{
-    /// Base Transcript
-    pub transcript: Transcript,
-
-    /// Type Parameter Marker
-    __: PhantomData<E>,
-}
-
-impl<E> TranscriptWrapper<E>
-where
-    E: PairingEngine,
-{
-    /// Builds a new [`TranscriptWrapper`] with the given `label`.
-    #[inline]
-    pub fn new(label: &'static [u8]) -> Self {
-        Self {
-            transcript: Transcript::new(label),
-            __: PhantomData,
-        }
-    }
-}
-
 /// Transcript adds an abstraction over the Merlin transcript
 /// For convenience
-pub(crate) trait TranscriptProtocol<E>
-where
-    E: PairingEngine,
-{
-    /// Append a `commitment` with the given `label`.
-    fn append_commitment(&mut self, label: &'static [u8], comm: &Commitment<E>);
+pub(crate) trait TranscriptProtocol {
+    /// Append an `item` with the given `label`.
+    fn append(&mut self, label: &'static [u8], item: &impl CanonicalSerialize);
```

Comment:
That's a really clever idea! I love it!

Reply:
Yeah this is one of the cases where working with the type system makes it much nicer/easier.